diff --git a/node_modules/react-native-fs/RNFSManager.m b/node_modules/react-native-fs/RNFSManager.m
index c443d20..78f130b 100755
--- a/node_modules/react-native-fs/RNFSManager.m
+++ b/node_modules/react-native-fs/RNFSManager.m
@@ -882,9 +882,15 @@ + (BOOL)requiresMainQueueSetup
     if ([asset isKindOfClass:[AVURLAsset class]]) {
       NSURL *url = [(AVURLAsset *)asset URL];
       NSLog(@"Final URL %@",url);
-      NSData *videoData = [NSData dataWithContentsOfURL:url];
-
-      BOOL writeResult = [videoData writeToFile:destination options:NSDataWritingAtomic error:&error];
+      BOOL writeResult = false;
+        
+      if (@available(iOS 9.0, *)) {
+          NSURL *destinationUrl = [NSURL fileURLWithPath:destination relativeToURL:nil];
+          writeResult = [[NSFileManager defaultManager] copyItemAtURL:url toURL:destinationUrl error:&error];
+      } else {
+          NSData *videoData = [NSData dataWithContentsOfURL:url];
+          writeResult = [videoData writeToFile:destination options:NSDataWritingAtomic error:&error];
+      }
         
       if(writeResult) {
         NSLog(@"video success");
