diff --git a/node_modules/react-native-fs/RNFSManager.m b/node_modules/react-native-fs/RNFSManager.m
index c443d20..2627e7f 100755
--- a/node_modules/react-native-fs/RNFSManager.m
+++ b/node_modules/react-native-fs/RNFSManager.m
@@ -881,17 +881,9 @@ + (BOOL)requiresMainQueueSetup
 
     if ([asset isKindOfClass:[AVURLAsset class]]) {
       NSURL *url = [(AVURLAsset *)asset URL];
+        NSURL *destinationURL = [NSURL fileURLWithPath:destination relativeToURL:nil];
       NSLog(@"Final URL %@",url);
-      NSData *videoData = [NSData dataWithContentsOfURL:url];
-
-      BOOL writeResult = [videoData writeToFile:destination options:NSDataWritingAtomic error:&error];
-
-      if(writeResult) {
-        NSLog(@"video success");
-      }
-      else {
-        NSLog(@"video failure");
-      }
+      [[NSFileManager defaultManager] copyItemAtURL:url toURL:destinationURL error:&error];
       dispatch_group_leave(group);
     }
   }];
